### PR TITLE
Jenkins modyfied by imaki

### PR DESCRIPTION
**Pull request by vecinomio**

### CODE REVIEW

File: ops/cloudformation/jenkins.yml [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+#  WaitHandle:
+#    Type: AWS::CloudFormation::WaitConditionHandle
+
+#  WaitCondition:
+#    Type: AWS::CloudFormation::WaitCondition
+#    DependsOn: "MyServerGroup"
+#    Properties:
+#      Handle: !Ref WaitHandle
+#      Timeout: "300"
+#      Count: 2
```
Comment (IYermakov) [inline ops/cloudformation/jenkins.yml]: Remove unused stuff

File: ops/cloudformation/jenkins.yml [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+# 1. Check VPC Stack. It must be up:
+#    aws cloudformation describe-stacks --stack-name ${VPCStackName}
+# 2. Check alb Stack. It must be up:
+#    aws cloudformation describe-stacks --stack-name ${albStackName}
+# 3. Check ebs-volume Stack. It must be up:
+#    aws cloudformation describe-stacks --stack-name ${ebsVolumeStackName}
+# 4. Validate jenkins template:
+#    aws cloudformation validate-template --template-body \
+#    file://ops/cloudformation/jenkins.yml
+# 5. Create ALB Stack:
```
Comment (IYermakov) [inline ops/cloudformation/jenkins.yml]: Jenkins, not ALB

File: ops/cloudformation/jenkins.yml (hunk 1 above)
Comment (vecinomio) [reply ops/cloudformation/jenkins.yml]: done

File: ops/cloudformation/jenkins.yml (hunk 2 above)
Comment (vecinomio) [reply ops/cloudformation/jenkins.yml]: done

File: ops/cloudformation/ebs-volume.yml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+          - AZ: !Select [ 0, !FindInMap [ "AZRegions", !Ref "AWS::Region", "PrivateAZs" ] ]
+      Encrypted: No
+      Size: !Ref Size
+      Tags:
+        - Key: "Name"
+          Value: "Jenkins-ebs"
+        - Key: "Service"
+          Value: "Jenkins"
+        - Key: "Environment"
+          Value: "ci"
```
Comment (Emperorex) [inline ops/cloudformation/ebs-volume.yml]: Actually it is hardcode. Would you use `Value: !Ref Environment`

File: ops/cloudformation/ebs-volume.yml [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        Fn::Sub:
+          - "${AWS::Region}${AZ}"
+          - AZ: !Select [ 0, !FindInMap [ "AZRegions", !Ref "AWS::Region", "PrivateAZs" ] ]
+      Encrypted: No
+      Size: !Ref Size
+      Tags:
+        - Key: "Name"
+          Value: "Jenkins-ebs"
+        - Key: "Service"
+          Value: "Jenkins"
```
Comment (Emperorex) [inline ops/cloudformation/ebs-volume.yml]: Actually it is hardcode. Would you use `Value: !Ref Service`

File: ops/cloudformation/install_0.0.1.sh [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+  local snapshot_id=$(wait_until aws ec2 describe-snapshots                                                   \
+                        --region ${region}                                                                    \
+                        --filters Name=status,Values=completed                                                \
+                                  Name=tag:Service,Values=Jenkins                                             \
+                        --query 'Snapshots[].[SnapshotId,StartTime] | reverse(sort_by(@, &[1])) | [0] | [0]' \
+                        --output text)
+
+  if [[ ${snapshot_id} = "None" || -z ${snapshot_id} ]]; then
+    log "Valid snapshot not found."
+    echo ""
```
Comment (Emperorex) [inline ops/cloudformation/install_0.0.1.sh]: Lets get rid of extra `echo`

File: ops/cloudformation/install_0.0.1.sh [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+  local volume_id=$(wait_until aws ec2 describe-volumes                        \
+                         --region ${region}                                    \
+                         --filters Name=status,Values=available                \
+                                   Name=tag:Service,Values=Jenkins             \
+                         --query 'Volumes[].[VolumeId,CreateTime] | reverse(sort_by(@, &[1])) | [0] | [0]'          \
+                         --output text)
+
+  if [[ ${volume_id} = "None" || -z ${volume_id} ]]; then
+    log "Valid volume not found."
+    echo ""
```
Comment (Emperorex) [inline ops/cloudformation/install_0.0.1.sh]: Lets get rid of extra `echo`

File: ops/cloudformation/jenkins.yml [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+      MinSize: '1'
+      MaxSize: '1'
+      TargetGroupARNs:
+        - Fn::ImportValue: alb-tgJenkins-ARN
+      Tags:
+        - Key: 'Name'
+          Value: 'Jenkins'
+          PropagateAtLaunch: true
+        - Key: 'Environment'
+          Value: 'CI'
```
Comment (Emperorex) [inline ops/cloudformation/jenkins.yml]: I think environment could be extracted as parameter.

File: ops/cloudformation/ebs-volume.yml [hunk 8]
```diff
@@ -0,0 +1,119 @@
+# ----------------------------- Instructions: ---------------------------------
+# 1. Check VPC Stack. It must be up:
```
Comment (mike-unitskyi) [inline ops/cloudformation/ebs-volume.yml]: extract this to readme

File: ops/cloudformation/ebs-volume.yml [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    Type: Number
+    Description: Size of volume in GB
+    Default: 8
+
+  Service:
+    Type: String
+    Description: lifecycle policy tag
+    Default: jenkins
+
+  Environment:
```
Comment (mike-unitskyi) [inline ops/cloudformation/ebs-volume.yml]: this parameter was introduced only in this CFN, does it makes sense to add it to other PRs?

File: ops/cloudformation/install_0.0.1.sh (hunk 6 above)
Comment (mike-unitskyi) [reply ops/cloudformation/install_0.0.1.sh]: i believe it's function output

File: ops/cloudformation/install_0.0.1.sh [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+  wait_until yum install jenkins -y
+  sed -i 's/JENKINS_ENABLE_ACCESS_LOG="\w*"/JENKINS_ENABLE_ACCESS_LOG="yes"/' /etc/sysconfig/jenkins
+  systemctl enable jenkins && systemctl start jenkins
+}
+
+#-------------------------------------------------------------------------------
+# Set hostname, mount or create volume, install Jenkins and additional packages.
+#-------------------------------------------------------------------------------
+function main() {
+  local region=$(get_ec2_instance_region)
```
Comment (mike-unitskyi) [inline ops/cloudformation/install_0.0.1.sh]: would move to `mount_jenkins_data_volume` function

File: ops/cloudformation/jenkins.yml [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+                  - ec2:DeleteSecurityGroup
+
+  InstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Path: "/"
+      Roles:
+        - Ref: InstanceRole
+
+  SimpleConfig:
```
Comment (mike-unitskyi) [inline ops/cloudformation/jenkins.yml]: would rename

File: ops/cloudformation/jenkins.yml [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+          - |
+            #!/bin/bash
+            set -o errexit
+            set -o nounset
+            set -o xtrace
+            set -o pipefail
+            PATH=$PATH:/opt/aws/bin/
+            aws s3 cp "s3://${S3Path}/scripts/jenkins/install_${ver}.sh" - | \
+            bash -l -s -- --hostname jenkins
+            sleep 1m
```
Comment (mike-unitskyi) [inline ops/cloudformation/jenkins.yml]: is this really needed?

File: ops/cloudformation/jenkins.yml [hunk 13]
```diff
@@ -0,0 +1,181 @@
+# ----------------------------- Instructions: ---------------------------------
+# 1. Check VPC Stack. It must be up:
```
Comment (mike-unitskyi) [inline ops/cloudformation/jenkins.yml]: would extract to readme

File: ops/cloudformation/jenkins.yml (hunk 12 above)
Comment (IYermakov) [reply ops/cloudformation/jenkins.yml]: > is this really needed?

I experienced a delay on starting jenkins service before I can do it healthcheck.

File: ops/cloudformation/ebs-volume.yml (hunk 8 above)
Comment (vecinomio) [reply ops/cloudformation/ebs-volume.yml]: done

File: ops/cloudformation/ebs-volume.yml (hunk 4 above)
Comment (vecinomio) [reply ops/cloudformation/ebs-volume.yml]: agree

File: ops/cloudformation/ebs-volume.yml (hunk 3 above)
Comment (vecinomio) [reply ops/cloudformation/ebs-volume.yml]: agree

File: ops/cloudformation/jenkins.yml (hunk 13 above)
Comment (vecinomio) [reply ops/cloudformation/jenkins.yml]: done

File: ops/cloudformation/jenkins.yml (hunk 7 above)
Comment (vecinomio) [reply ops/cloudformation/jenkins.yml]: done

File: ops/cloudformation/install_0.0.1.sh (hunk 5 above)
Comment (vecinomio) [reply ops/cloudformation/install_0.0.1.sh]: done

File: ops/cloudformation/jenkins.yml (hunk 11 above)
Comment (vecinomio) [reply ops/cloudformation/jenkins.yml]: done

**Pull request closed without merging**